src/geventhttpclient/tests/test_ssl.py

```diff
@@ -19,6 +20,8 @@
 
 @contextmanager
 def server(handler, backlog=1):
+    if six.PY3:
+        gevent.sleep(10)
```

Comment:
I have the suspicion that [Greenlet.kill](http://www.gevent.org/gevent.html#gevent.Greenlet.kill) is invoked on Python3 to a different time point than on Python2.